### PR TITLE
Fix deprecation warnings.

### LINKED ISSUE #17: DeprecationWarning: ComponentLookupError is deprecated.
```
zope/pagetemplate/engine.py:24: DeprecationWarning: ComponentLookupError is deprecated. Import from zope.interface.interfaces
  from zope.component.interfaces import ComponentLookupError
```

### PR DESCRIPTION
**Pull request by jamadden**
Fixes #17

### CODE REVIEW

File: src/zope/pagetemplate/tests/test_engine.py [hunk 1]
```diff
@@ -86,6 +86,9 @@ def test_disallowed_builtin(self):
 class TestZopeContext(PlacelessSetup,
                       unittest.TestCase):
 
+    assertRaisesRegex = getattr(unittest.TestCase, 'assertRaisesRegex',
+                                getattr(unittest.TestCase, 'assertRaisesRegexp'))
```
Comment (ale-rt) [inline src/zope/pagetemplate/tests/test_engine.py]: This is is always calling getattr(unittest.TestCase, 'assertRaisesRegexp').
Would not be better to:
```
try:
     assertRaisesRegex = unittest.TestCase.assertRaisesRegex
except Attribute Error:
     # Support X version of Y
     assertRaisesRegex = unittest.TestCase.assertRaisesRegexp
```

Not a blocker anyway.

Comment (jamadden) [reply src/zope/pagetemplate/tests/test_engine.py]: I write it that way basically to cheat: There's no implicit branch, all statements are always executed, and coverage is unaffected.

**Comment by jamadden** [comment]
Thanks!

File: src/zope/pagetemplate/tests/test_engine.py (hunk 1 above)
Comment (ale-rt) [reply src/zope/pagetemplate/tests/test_engine.py]: I see your point :)

Comment (mgedmin) [reply src/zope/pagetemplate/tests/test_engine.py]: Isn't this equivalent to 

```python
    assertRaisesRegex = getattr(unittest.TestCase, 'assertRaisesRegex',
                                unittest.TestCase.assertRaisesRegexp)
```

?

Why the second `getattr()`?

Comment (jamadden) [reply src/zope/pagetemplate/tests/test_engine.py]: Yes it is. I have no idea why I wrote it that way 😢  [In the past](https://github.com/zopefoundation/ZConfig/blob/master/ZConfig/tests/support.py#L63) I've written it directly.

**Pull request merged**